libcudacxx/include/cuda/std/detail/libcxx/include/__algorithm/comp.h

```diff
@@ -0,0 +1,57 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCUDACXX___ALGORITHM_COMP_H
+#define _LIBCUDACXX___ALGORITHM_COMP_H
+
+#ifndef __cuda_std__
+#  include <__config>
+#endif // __cuda_std__
+
+#include "../__type_traits/integral_constant.h"
+#if defined(_LIBCUDACXX_HAS_STRING)
+#  include "../__type_traits/predicate_traits.h"
+#endif // _LIBCUDACXX_HAS_STRING
+
+#if defined(_CCCL_COMPILER_NVHPC) && defined(_CCCL_USE_IMPLICIT_SYSTEM_DEADER)
+#  pragma GCC system_header
+#else // ^^^ _CCCL_COMPILER_NVHPC ^^^ / vvv !_CCCL_COMPILER_NVHPC vvv
+_CCCL_IMPLICIT_SYSTEM_HEADER
+#endif // !_CCCL_COMPILER_NVHPC
+_LIBCUDACXX_BEGIN_NAMESPACE_STD
+
+struct __equal_to
+{
+  template <class _T1, class _T2>
+  _LIBCUDACXX_NODISCARD_EXT _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_INLINE_VISIBILITY _LIBCUDACXX_CONSTEXPR_AFTER_CXX11 bool
+  operator()(const _T1& __lhs, const _T2& __rhs) const noexcept(noexcept(__lhs == __rhs))
```

Comment:
suggestion: in upstream and standard these functions do not use noexcept specifier. This can have visible side effects:
```
struct equal_to_1 {
  template <class _T1, class _T2>
  bool operator()(const _T1& __lhs, const _T2& __rhs) const noexcept(noexcept(__lhs == __rhs)) {
    return __lhs == __rhs;
  }
};

struct equal_to_2 {
  template <class _T1, class _T2>
  bool operator()(const _T1& __lhs, const _T2& __rhs) const {
    return __lhs == __rhs;
  }
};

int main() {
  static_assert(noexcept(equal_to_1{}(1, 2)));
  static_assert(noexcept(equal_to_2{}(1, 2)));
} 
```

Unless there's a strong reason to deviate from upstream, I wouldn't do that (same for `__less`).

Reply:
Generally, adding noexcept specifiers is allowed by the standard an QoI detial.

My point is that libc++ is not good at it and we should be better, even if we do not care about exceptions on device